Lib/test/libregrtest/refleak.py

```diff
@@ -7,6 +7,13 @@
 from test import support
 
 
+def maybe_small_long(x):
+    # bpo-31217: On CPython, x_sub() of longobject.c doesn't try to use int
+    # singletons. Try to get singletons to prevent false alarms on memory block
+    # leaks.
+    return int(str(x))
```

Comment:
Rather than rely on small int singletons, I think an explicit pool would be less implementation-dependent.